src/content/content.js

```diff
@@ -36,3 +37,7 @@ export const mountContentScript = (contentScript) => {
     console.log('Overlay is finished');
   });
 };
+
+export const reloadWhenURLChanged = (callback) => {
+  addMessagingEventListener(EVENT_URL_CHANGED, callback);
+};
```

Comment:
Here, can you listen to `browser.runtime.onMessage` instead of `window.onmessage`?

Reply:
AFAIK you cannot add two `onMessage` listeners... so if you defined this listener in `content-events.js` you won't be able to add another listener in another file

Reply:
I think this PR should be finished. You're welcome to suggest your way in another PR.